src/SWP/Bundle/CoreBundle/Controller/AnalyticsExportController.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SWP\Bundle\CoreBundle\Controller;
+
+use Nelmio\ApiDocBundle\Annotation\Operation;
+use Nelmio\ApiDocBundle\Annotation\Model;
+use Swagger\Annotations as SWG;
+use DateTime;
+use Doctrine\Common\Cache\Cache;
+use Hoa\Mime\Mime;
+use League\Flysystem\Filesystem;
+use SWP\Bundle\CoreBundle\AnalyticsExport\CsvReportFileLocationResolver;
+use SWP\Bundle\CoreBundle\AnalyticsExport\ExportAnalytics;
+use SWP\Bundle\CoreBundle\Context\CachedTenantContextInterface;
+use SWP\Bundle\CoreBundle\Model\AnalyticsReport;
+use SWP\Bundle\CoreBundle\Model\AnalyticsReportInterface;
+use SWP\Bundle\CoreBundle\Model\UserInterface;
+use SWP\Component\Common\Criteria\Criteria;
+use SWP\Component\Common\Pagination\PaginationData;
+use SWP\Component\Common\Response\ResourcesListResponse;
+use SWP\Component\Storage\Repository\RepositoryInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+use Symfony\Component\Routing\Annotation\Route;
+use SWP\Component\Common\Response\ResponseContext;
+use SWP\Component\Common\Response\SingleResourceResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+class AnalyticsExportController extends AbstractController
+{
+    protected $cacheProvider;
+
+    protected $analyticsReportRepository;
+
+    protected $filesystem;
+
+    protected $csvReportFileLocationResolver;
+
+    protected $cachedTenantContext;
+
+    public function __construct(
+        Cache $cacheProvider,
+        RepositoryInterface $analyticsReportRepository,
+        Filesystem $filesystem,
+        CsvReportFileLocationResolver $csvReportFileLocationResolver,
+        CachedTenantContextInterface $cachedTenantContext
+    ) {
+        $this->cacheProvider = $cacheProvider;
+        $this->analyticsReportRepository = $analyticsReportRepository;
+        $this->filesystem = $filesystem;
+        $this->csvReportFileLocationResolver = $csvReportFileLocationResolver;
+        $this->cachedTenantContext = $cachedTenantContext;
+    }
+
+    /**
+     * @Operation(
+     *     tags={"analytics"},
+     *     summary="Export analytics data"
+     * )
+     *
+     * @Route("/api/{version}/export/analytics", options={"expose"=true}, defaults={"version"="v2"}, methods={"POST"}, name="swp_api_core_analytics_export_post")
+     *
+     * @return SingleResourceResponse
+     *
+     * @throws \Exception
+     */
+    public function post(Request $request): SingleResourceResponse
+    {
+        /** @var UserInterface $currentlyLoggedInUser */
+        $currentlyLoggedInUser = $this->getUser();
+
+        $start = new DateTime($request->query->get('start', 'now'));
```

Comment:
beside `$fileName` all those variables can be removed and values provided in `ExportAnalytics` constructor.

Reply:
why?

Reply:
they takes lines, and memory and can be safely provided directly in contructor (in one place)

Reply:
but it's a micro-optimization which makes no sense here